src/ids-host.ts

```diff
@@ -31,6 +35,63 @@ export class IdsHost {
     this.idsProjectName = idsProjectName;
     this.diagnosticsSuffix = diagnosticsSuffix;
     this.runtimeDiagnosticsUrl = runtimeDiagnosticsUrl;
+    this.client = undefined;
+  }
+
+  async getGot(
+    recordFailoverCallback?: (prevUrl: URL, nextUrl: URL) => void,
```

Comment:
Is there a reason to make the callback configurable?

Reply:
Makes it easier for testing, and also we have to have a relationship between the main action class and this one. That gets a bit awkward, and it is nicer to let that side manage that sid'sbusiness and just pass in a hook.